src/sinks/redis.rs

```diff
@@ -0,0 +1,522 @@
+use crate::{
+    buffers::Acker,
+    config::{log_schema, DataType, GenerateConfig, SinkConfig, SinkContext, SinkDescription},
+    event::Event,
+    internal_events::{
+         RedisEventSent, RedisSendEventFailed, RedisEncodeEventFailed, TemplateRenderingFailed,
+    },
+    sinks::util::encoding::{EncodingConfig, EncodingConfigWithDefault, EncodingConfiguration},
+    template::{Template, TemplateParseError},
+};
+
+use futures::{future::BoxFuture, ready, stream::FuturesUnordered, FutureExt, Sink, Stream};
+use redis::{aio::ConnectionManager, AsyncCommands, RedisError, RedisResult};
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use std::{
+    collections::HashSet,
+    convert::TryFrom,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+#[derive(Debug, Snafu)]
+enum BuildError {
+    #[snafu(display("creating Redis producer failed: {}", source))]
+    RedisCreateFailed { source: RedisError },
+    #[snafu(display("invalid key template: {}", source))]
+    KeyTemplate { source: TemplateParseError },
+}
+
+#[derive(Clone, Debug, Derivative, Deserialize, Serialize)]
+#[derivative(Default)]
+#[serde(rename_all = "lowercase")]
+pub enum Type {
+    #[derivative(Default)]
+    List,
+    Channel,
+}
+
+#[derive(Clone, Copy, Debug, Derivative, Deserialize, Serialize, Eq, PartialEq)]
+#[derivative(Default)]
+#[serde(rename_all = "lowercase")]
+pub enum Method {
+    #[derivative(Default)]
```

Comment:
```suggestion
#[serde(rename_all = "lowercase")]
pub enum Method {
```

According to https://github.com/timberio/vector/pull/7078/files#r618892597, the default is never used, correct?

Reply:
just because  I choices `list` as the default `data_type`, so  I also choices a `method` (`lpush` or `rpush`) as the default value